pkg/config/remote/uptane/partial_client_test.go

```diff
@@ -149,9 +151,164 @@ func TestPartialClientVerifyInvalidRotation(t *testing.T) {
 	testRepository2 := newTestRepository(2, nil, targets1, []*pbgo.File{{Path: "datadog/2/APM_SAMPLING/id/1", Raw: target1content}})
 
 	err = client.Update(testRepository2.toPartialUpdate())
+	// TODO in the whole file: use "assert.ErrorAs" with specific error type
 	assert.Error(t, err)
 	_, err = client.Targets()
 	assert.Error(t, err)
 	_, err = client.TargetFile("datadog/2/APM_SAMPLING/id/1")
 	assert.Error(t, err)
 }
+
+func TestPartialClientRootKeyRotation(t *testing.T) {
+	_, targetFileMeta := generateTarget()
+	directorTargetMetadata := data.TargetFiles{
+		"datadog/2/APM_SAMPLING/id/1": targetFileMeta,
+	}
+
+	repository1 := newTestRepository(1, nil, directorTargetMetadata, nil)
+	config.Datadog.Set("remote_configuration.director_root", repository1.directorRoot)
+	config.Datadog.Set("remote_configuration.config_root", repository1.configRoot)
+
+	client, err := NewPartialClient()
+	require.NoError(t, err)
+
+	err = client.Update(repository1.toPartialUpdate())
+	require.NoError(t, err)
+
+	repository2 := newTestRepository(2, nil, directorTargetMetadata, nil)
+	repository2.directorRootVersion = repository1.directorRootVersion + 1
+	repository2.directorRoot = generateRoot(
+		repository2.directorRootKey,
+		repository2.directorRootVersion,
+		repository2.directorTimestampKey,
+		repository2.directorTargetsKey,
+		repository2.directorSnapshotKey,
+		// new root must be signed by old root
+		repository1.directorRootKey,
+	)
+
+	err = client.Update(repository2.toPartialUpdate())
+	require.NoError(t, err)
+
+	root, err := client.getRoot()
+	require.NoError(t, err)
+	assert.Equal(t, root.Roles["root"].KeyIDs[0], repository2.directorRootKey.PublicData().IDs()[0])
+}
+
+// TestPartialClientRejectsUnsignedTarget tests that the partial uptane client
+// does not accept targets which are not listed in the targets metadata file
+func TestPartialClientRejectsUnsignedTarget(t *testing.T) {
+	files := []*pbgo.File{
+		{Path: "datadog/2/APM_SAMPLING/id/1", Raw: []byte("mAlIcIoUs cOnTeNt!!")},
```

Comment:
can you change `1` to `config`? that's the only configuration file name we have and tests should be as close as possible to reality

Reply:
This was copy-pasted from other parts of this file, should I replace `1` with `config` in every other occurrence in this file as well?

Reply:
I won't ask you to do that, if you want you can

Reply:
Done in 0f09576